Wuziqi

f8daniel opened the pull request:
- Major overhaul of the GUI, adding a wooden board effect (but placing tiles in the center of squares instead of the hatch marks like in Go).
- Cleaned up messaging to the user
- Added explanatory comments, reordered some code, and changed a couple function names to be clearer as a reference to developers.

Remaining Issues:

1. There was a bug with the end game scenario, which throw an error. Temporary patch by commenting out the offending line and leaving it up to the players to manually exit the game (instead of calling resignGame).
2. Also seeing a (nonfatal) error in the arcade during launch, with an undefined reference to boardElement.
3. Annoys me that the entire board is being sent back and forth between players instead of just shorthand notation (cell+player_id), but maybe we want to be inefficient for some reason?

arpee commented:
_* Major overhaul of the GUI, adding a wooden board effect (but placing tiles in the center of squares instead of the hatch marks like in Go)._

Wuziqi is played on a go board with pieces placed on the intersections of the lines not in the boxes. I am also not a big fan of skeuomorphic design possibly to the point of prejudice, so would prefer to drop this.

_* Cleaned up messaging to the user_ 
This is much better, thanks.

_* Added explanatory comments, reordered some code, and changed a couple function names to be clearer as a reference to developers._ 
Also an improvement, thanks.

On separate matters - use of the dataset feature can be useful, and better in cases where you are loading data into cells, like an id that needs to be passed back to a db but should not be the element id, for example. Here I feel it is the wrong way to go, as this information is not read by the javascript, so it means we have to use largely unfamiliar CSS to access the data, and, we have to look in multiple places for this information. Nothing wrong with multiple classes...

Regarding the CSS at lines 74~83, this was a hack to make it possible to click on the area around the visible point of the cell. (Because cell margins are not clickable, this is a way of bubbling the click event for a bigger area to register for the content. This can be useful for small things in the ui generally. This could be lost completely in your gui rework.)

Regarding sending the board back and fourth, this is optional. For this game, given it's size, it's not overly harmful and comes with the benefit that we don't need to maintain any state. Get the latest move alone, and the game updates correctly.

If we were only sending incremental moves, this would save space, but we would need to replay the whole game from stored transactions (the client won't download the whole chain to find old moves) or store state in the wallet. The second is a feasible approach, but adds co-ordination problems to handle issues where a move is missed etc. For simple games, partially designed to teach folks how to code a game.

One other thing I note is some updates to potentially allow for p>2 games. This is not problematic - but having things spelled out as two player was intentional on my part, for clarity and simplicity. (I thought about generalising to multi players, but the game really does not work. Teams might...)

There are also a few oddities, like having the board background for the board and each space. And, some real improvements in the cleanup. Having said that  Let me think on and poll the design changes with folks - see which is preferred.

f8daniel commented:
> Wuziqi is played on a go board with pieces placed on the intersections of the lines not in the boxes. I am also not a big fan of skeuomorphic design possibly to the point of prejudice, so would prefer to drop this.

So, are you annoyed that I am trying to make it look like a Go board or that it doesn't look enough like a Go board? I already have two ideas for fixing the placement to look like a Go board, but I really think the wooden pattern on our standard green felt looks nicer than the (saved fallback css) color gradients.

I can change the dataset back to classes in that one case.

Finally, I didn't mention it, but after I changed over the css logic, I did test on very small screen sizes and I didn't encounter any problems clicking on cells like you mentioned, so *fingers crossed* I inadvertently solved that bug.

arpee commented:
Not annoyed,  I just really prefer abstract designs for games to mimicking
a non-digital version.  I'm going to put it to the office tomorrow - they
will likely come down on the 'look like real thing' option.

 The issue with the small click target was distinct to the way I was doing
the dots to click on. By making them smaller (the big margins), they were
rough to click. With the squares not having border margins, that would not
be an issue at all.

There will be a way of having two game grids, on for the pieces and one for
the grid, would not be too difficult. Just need to do the shifting of one
under the other in the same units (i.e. vw) as board size is defined.

R

Richard Parris - Co-Founder

ph : +86 1860 1221 974


On Wed, 8 Sept 2021 at 21:35, Daniel Worlton ***@***.***>
wrote:

> Wuziqi is played on a go board with pieces placed on the intersections of
> the lines not in the boxes. I am also not a big fan of skeuomorphic design
> possibly to the point of prejudice, so would prefer to drop this.
>
> So, are you annoyed that I am trying to make it look like a Go board or
> that it doesn't look enough like a Go board? I already have two ideas for
> fixing the placement to look like a Go board, but I really think the wooden
> pattern on our standard green felt looks nicer than the (saved fallback
> css) color gradients.
>
> I can change the dataset back to classes in that one case.
>
> Finally, I didn't mention it, but after I changed over the css logic, I
> did test on very small screen sizes and I didn't encounter any problems
> clicking on cells like you mentioned, so *fingers crossed* I
> inadvertently solved that bug.
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/SaitoTech/saito-lite/pull/209#issuecomment-915244955>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAV4EQ5PPAZ3GFGTQO56KPTUA5RB3ANCNFSM5DTOCCJA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

f8daniel commented:
1. Found a bug where both players had enabled events at beginning of game.

2. Improved CSS to handle mobile devices in portrait/landscape mode, and caught a bug where small landscape mode made the game 10x bigger than the screen.

3. Implemented two solutions to lay down the lines like a Go board, so there are lots of commented out lines in the css file. Going with a manually drawn line through a tile image for now, although different browsers/screen sizes can create a funky visual effect as the tile is scaled to fit the board space. On the other hand, using a ::after pseudo-element to draw the grid gives clean lines but makes it very difficult to give the board an evenly sized rim. Btw, the reason the board itself has an image-background is that it is an easy way to fill in a thick outline around the playable board. It's merely an aesthetic thing and we can shrink or eliminate the padding as desired, or sub in a different wood color/texture to look like an even more expensive game board (picture a fancy chessboard with a mahogany frame inlaid with two different kinds of marble) . I find natural looking textures/colors more pleasing to the eye.

4. Reverted from datasets for black/white and playerscore back to classes.

5. Currently, the .active:hover makes a nice size semi-transparent circle to help visualize the click target.  Even a 200px by 300px is easy to click on, though it does raise questions about game-menu
![image](https://user-images.githubusercontent.com/70703104/132593466-a115c6f5-68c4-4e70-a61b-fcbcc142c5ce.png)

arpee commented:
Ahoy - I have stolen some of the fixes but will hold off on pusing as a whole.

Kekya the in house designer has volunteered to work up some ui elements for the scores etc.

Might make sense to incorp before pushing this. Will see what he comes up with. 

IF nothing before Friday will push this (good to announce.)

Thanks man.

f8daniel commented:
What? Are you suggesting there is some sort of team collaboration on this? That's crazy talk. 
Look forward to seeing what y'all come up with, but I don't envy you the task of merging multiple, potentially conflicting branches

arpee commented:
I know/...

Richard Parris - Co-Founder

ph : +86 1860 1221 974


On Tue, 14 Sept 2021 at 22:36, Daniel Worlton ***@***.***>
wrote:

> What? Are you suggesting there is some sort of team collaboration on this?
> That's crazy talk.
> Look forward to seeing what y'all come up with, but I don't envy you the
> task of merging multiple, potentially conflicting branches
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/SaitoTech/saito-lite/pull/209#issuecomment-919212903>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAV4EQZYBFV7HVNAH7OWGW3UB5MVZANCNFSM5DTOCCJA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>
